src/net_processing.cpp

```diff
@@ -2051,8 +2088,9 @@ void PeerManagerImpl::ProcessGetData(CNode& pfrom, Peer& peer, const std::atomic
 
         const CInv &inv = *it++;
 
-        if (peer.m_tx_relay == nullptr) {
-            // Ignore GETDATA requests for transactions from blocks-only peers.
+        if (tx_relay == nullptr) {
+            // Ignore GETDATA requests for transactions from blocks-only peers and
```

Comment:
```suggestion
            // Ignore GETDATA requests for transactions from block-relay-only peers and
```

nit in the last commit: Maybe clarify while touching this line